README.md
# douglasie README

This is the README for your extension "douglasie". After writing up a brief description, we recommend including the following sections.

## Features

Describe specific features of your extension including screenshots of your extension in action. Image paths are relative to this README file.

For example if there is an image subfolder under your extension project workspace:

\!\[feature X\]\(images/feature-x.png\)

> Tip: Many popular extensions utilize animations. This is an excellent way to show off your extension! We recommend short, focused animations that are easy to follow.

## Requirements

If you have any requirements or dependencies, add a section describing those and how to install and configure them.

## Extension Settings

Include if your extension adds any VS Code settings through the `contributes.configuration` extension point.

For example:

This extension contributes the following settings:

* `myExtension.enable`: enable/disable this extension
* `myExtension.thing`: set to `blah` to do something

## Known Issues

Calling out known issues can help limit users opening duplicate issues against your extension.

## Release Notes

Users appreciate release notes as you update your extension.

### 1.0.0

Initial release of ...

### 1.0.1

Fixed issue #.

### 1.1.0

Added features X, Y, and Z.

-----------------------------------------------------------------------------------------------------------
## Following extension guidelines

Ensure that you've read through the extensions guidelines and follow the best practices for creating your extension.

* [Extension Guidelines](https://code.visualstudio.com/api/references/extension-guidelines)

## Working with Markdown

**Note:** You can author your README using Visual Studio Code.  Here are some useful editor keyboard shortcuts:

* Split the editor (`Cmd+\` on macOS or `Ctrl+\` on Windows and Linux)
* Toggle preview (`Shift+CMD+V` on macOS or `Shift+Ctrl+V` on Windows and Linux)
* Press `Ctrl+Space` (Windows, Linux) or `Cmd+Space` (macOS) to see a list of Markdown snippets

### For more information

* [Visual Studio Code's Markdown Support](http://code.visualstudio.com/docs/languages/markdown)
* [Markdown Syntax Reference](https://help.github.com/articles/markdown-basics/)

**Enjoy!**


CHANGELOG.md
# Change Log

All notable changes to the "douglasie" extension will be documented in this file.

Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.

## [Unreleased]

- Initial release

out/extension.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.deactivate = exports.activate = void 0;
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
const vscode = require("vscode");
const advanced_viewer_1 = require("./advanced-viewer");
const EXTENSION_NAME = 'Douglasie';
let editor;
// this method is called when your extension is activated
// your extension is activated the very first time the command is executed
function activate(context) {
    return __awaiter(this, void 0, void 0, function* () {
        // Use the console to output diagnostic information (console.log) and errors (console.error)
        // This line of code will only be executed once when your extension is activated
        console.log('Congratulations, your extension "douglasie" is now active!');
        editor = vscode.window.registerCustomEditorProvider('douglasie.parquet', new advanced_viewer_1.AdvancedViewer());
        // The command has been defined in the package.json file
        // Now provide the implementation of the command with registerCommand
        // The commandId parameter must match the command field in package.json
        let disposable = vscode.commands.registerCommand('douglasie.openParquet', () => {
            // The code you place here will be executed every time your command is executed
            // Display a message box to the user
            vscode.window.showInformationMessage('Hello World from Douglasie!');
        });
        context.subscriptions.push(disposable);
        context.subscriptions.push(editor);
    });
}
exports.activate = activate;
// this method is called when your extension is deactivated
function deactivate() {
    //editor.dispose();
}
exports.deactivate = deactivate;
//# sourceMappingURL=extension.js.map

src/extension.ts
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
import * as vscode from 'vscode';
import { AdvancedViewer } from './advanced-viewer';
import { DouglasieViewer } from './douglasie-viewer';


const EXTENSION_NAME = 'Douglasie';

let editor: vscode.Disposable;

// this method is called when your extension is activated
// your extension is activated the very first time the command is executed
export async function activate(context: vscode.ExtensionContext) {
	
	// Use the console to output diagnostic information (console.log) and errors (console.error)
	// This line of code will only be executed once when your extension is activated
	console.log('Congratulations, your extension "douglasie" is now active!');


	editor = vscode.window.registerCustomEditorProvider('douglasie.parquet', new AdvancedViewer());

	// The command has been defined in the package.json file
	// Now provide the implementation of the command with registerCommand
	// The commandId parameter must match the command field in package.json
	let disposable = vscode.commands.registerCommand('douglasie.openParquet', () => {
		// The code you place here will be executed every time your command is executed
		// Display a message box to the user


		vscode.window.showInformationMessage('Hello World from Douglasie!');
	});

	context.subscriptions.push(disposable);
	context.subscriptions.push(editor);
}

// this method is called when your extension is deactivated
export function deactivate() {
	//editor.dispose();
}


src/advanced-viewer.ts
import { CancellationToken, CustomDocument, CustomDocumentBackup, CustomDocumentBackupContext, CustomDocumentContentChangeEvent, CustomDocumentEditEvent, CustomDocumentOpenContext, CustomEditorProvider, CustomReadonlyEditorProvider, Event, Uri, WebviewPanel } from "vscode";


export class ParquetDocument implements CustomDocument {
    uri: Uri;

    constructor(uri: Uri) {
        this.uri = uri;
    }

    dispose(): void {
        
    }

    getContent(): string {
        return 'Hello World';
    }

}

export class AdvancedViewer implements CustomReadonlyEditorProvider<ParquetDocument> {
    openCustomDocument(uri: Uri, openContext: CustomDocumentOpenContext, token: CancellationToken): ParquetDocument | Thenable<ParquetDocument> {
        return new ParquetDocument(uri);
    }
    resolveCustomEditor(document: ParquetDocument, webviewPanel: WebviewPanel, token: CancellationToken): void | Thenable<void> {
        webviewPanel.webview.html = document.getContent();
    }



}

src/douglasie-viewer.ts
import * as vscode from 'vscode';

export class DouglasieViewer implements vscode.CustomTextEditorProvider {

    outputChannel;

    constructor() {
        this.outputChannel = vscode.window.createOutputChannel('douglasie');
        this.outputChannel.show(true);
        this.outputChannel.appendLine('Loading DouglasieViewer');
    }
    resolveCustomTextEditor(document: vscode.TextDocument, webviewPanel: vscode.WebviewPanel, token: vscode.CancellationToken): void | Thenable<void> {
        const content = document.getText();
        webviewPanel.title = 'DouglasieViewer';
        webviewPanel.webview.html = `
            <table>
                <tr>
                <th>Firstname</th>
                <th>Lastname</th>
                <th>Age</th>
                </tr>
                <tr>
                <td>Jill</td>
                <td>Smith</td>
                <td>50</td>
                </tr>
                <tr>
                <td>Eve</td>
                <td>Jackson</td>
                <td>94</td>
                </tr>
            </table> 
        `;
    }
}

out/douglasie-viewer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DouglasieViewer = void 0;
const vscode = require("vscode");
class DouglasieViewer {
    constructor() {
        this.outputChannel = vscode.window.createOutputChannel('douglasie');
        this.outputChannel.show(true);
        this.outputChannel.appendLine('Loading DouglasieViewer');
    }
    resolveCustomTextEditor(document, webviewPanel, token) {
        const content = document.getText();
        webviewPanel.title = 'DouglasieViewer';
        webviewPanel.webview.html = `
            <table>
                <tr>
                <th>Firstname</th>
                <th>Lastname</th>
                <th>Age</th>
                </tr>
                <tr>
                <td>Jill</td>
                <td>Smith</td>
                <td>50</td>
                </tr>
                <tr>
                <td>Eve</td>
                <td>Jackson</td>
                <td>94</td>
                </tr>
            </table> 
        `;
    }
}
exports.DouglasieViewer = DouglasieViewer;
//# sourceMappingURL=douglasie-viewer.js.map